src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -274,6 +274,14 @@ internal static unsafe bool ObjectHasComponentSize(object obj)
         [MethodImpl(MethodImplOptions.InternalCall)]
         internal static extern unsafe object? Box(MethodTable* methodTable, ref byte data);
 
+        /// <summary>
+        /// Allocates a new uninitialized object using an input <see cref="MethodTable"/> to determine its type.
+        /// </summary>
+        /// <param name="methodTable">The <see cref="MethodTable"/> pointer to use to create the instance.</param>
+        /// <returns>An instance of the type specified by <paramref name="methodTable"/>.</returns>
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern unsafe object AllocateUninitializedObject(MethodTable* methodTable);
```

Comment:
Can GetUninitializedObject now be implemented in terms of AllocateUninitializedObject rather than having its own QCall and native impl?

Reply:
It could, but it seems to me that would require some additional work:
- That API also has additional logic to validate the input type handle (in `RuntimeTypeHandle::ValidateTypeAbleToBeInstantiated`)
- Currently, the new `AllocateUninitializedObject` API will skip running static cctor (as that's not needed from the current callsite, as when an object is being cloned, the static cctor for that type has necessarily already been run). We'd need to port the logic to check that, so we might need a separate runtime call anyway.

We might potentially investigate that in a separate PR? 🙂